scripts/sct_dmri_separate_b0_and_dwi.py

```diff
@@ -40,84 +42,102 @@ def __init__(self):
-                      mandatory=True,
-                      example='dmri.nii.gz')
-    parser.add_option(name='-bvec',
-                      type_value='file',
-                      description='bvecs file',
-                      mandatory=True,
-                      example='bvecs.txt')
-
-    # Optional arguments
-    parser.add_option(name='-a',
-                      type_value='multiple_choice',
-                      description='average b=0 and DWI data.',
-                      mandatory=False,
-                      example=['0', '1'],
-                      default_value=str(param_default.average))
-    parser.add_option(name='-bval',
-                      type_value='file',
-                      description='bvals file. Used to identify low b-values (in case different from 0).',
-                      mandatory=False)
-    parser.add_option(name='-bvalmin',
-                      type_value='float',
-                      description='B-value threshold (in s/mm2) below which data is considered as b=0.',
-                      mandatory=False,
-                      example='50')
-    parser.add_option(name='-ofolder',
-                      type_value='folder_creation',
-                      description='Output folder.',
-                      mandatory=False,
-                      default_value='./')
-    parser.add_option(name='-v',
-                      type_value='multiple_choice',
-                      description='Verbose.',
-                      mandatory=False,
-                      example=['0', '1'],
-                      default_value=str(param_default.verbose))
-    parser.add_option(name='-r',
-                      type_value='multiple_choice',
-                      description='remove temporary files.',
-                      mandatory=False,
-                      example=['0', '1'],
-                      default_value=str(param_default.remove_temp_files))
-
+    parser = argparse.ArgumentParser(
+        description="Separate b=0 and DW images from diffusion dataset. The output files will have a suffix "
+                    "(_b0 and _dwi) appended to the input file name.",
+        formatter_class=SmartFormatter,
+        add_help=None,
+        prog=os.path.basename(__file__).strip(".py")
+    )
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        '-i',
+        metavar=Metavar.file,
+        required=True,
+        help="Diffusion data. Example: dmri.nii.gz"
+    )
+    mandatory.add_argument(
+        '-bvec',
+        metavar=Metavar.file,
+        required=True,
+        help="Bvecs file. Example: bvecs.txt"
+    )
+
+    optional = parser.add_argument_group("\nOPTIONAL ARGUMENTS")
+    optional.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        help="Show this help message and exit."
+    )
+    optional.add_argument(
+        '-a',
+        type=int,
+        metavar=Metavar.int,
+        choices=(0, 1),
```

Comment:
i think everytime the variable "choices" is set, we should not display "metavar" because it will overwrite the display of the available choices to the user